### PR TITLE
ID-2739: Aggregate TS annotation status upwards in hierarchy

### PR DESCRIPTION
**Pull request by erikmaland**

### CODE REVIEW

**Review by pettermk — CHANGES_REQUESTED** [review]
As we discussed it might be possible to optimize this to avoid frequently searching the `node_severity_` array. Also small comment regarding formatting

File: tag_hierarchy/unittests/test_severity_field.cpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                        {{
+                                 {
+                                         std::string("modelclassfilter"), std::vector<std::string>{"44380be8-76d2-a1fd-dc50-0ad8f4c151e4"}
+                                 }
+                         }}
+                ));
+        }
+
+BOOST_AUTO_TEST_SUITE_END()
+
```
[the remark this answers is not in the document]
Comment (erikmaland) [reply tag_hierarchy/unittests/test_severity_field.cpp]: Fixed editing of function to the established style.

**Comment by erikmaland** [comment]
> As we discussed it might be possible to optimize this to avoid frequently searching the `node_severity_` array. Also small comment regarding formatting

I did not manage to use the path-variable as we discussed. The reason being that the severity property is not written to it during the traversal.

Did get rid of one call to `find()` on the map. Also changed from `find()` to `count()` as this call does not construct an iterator.

**Review by pettermk — APPROVED** [review]
Good work!

**Pull request merged**